src/legacy/core_plugins/data/public/search/aggs/agg_config.test.ts

```diff
@@ -0,0 +1,499 @@
+      expect(state).toHaveProperty('schema', 'segment');
+    });
+
+    it('test serialization  order is identical (for visual consistency)', () => {
+      const configStates = [
+        {
+          enabled: true,
+          type: 'date_histogram',
+          schema: 'segment',
+          params: {},
+        },
+      ];
+      const ac1 = new AggConfigs(indexPattern, configStates, { typesRegistry });
+      const ac2 = new AggConfigs(indexPattern, configStates, { typesRegistry });
+
+      // this relies on the assumption that js-engines consistently loop over properties in insertion order.
+      // most likely the case, but strictly speaking not guaranteed by the JS and JSON specifications.
+      expect(JSON.stringify(ac1.aggs) === JSON.stringify(ac2.aggs)).toBe(true);
+    });
+  });
+
+  describe('#makeLabel', () => {
+    let aggConfig: AggConfig;
+
+    beforeEach(() => {
+      const ac = new AggConfigs(indexPattern, [], { typesRegistry });
+      aggConfig = ac.createAggConfig({ type: 'count' } as CreateAggConfigParams);
+    });
+
+    it('uses the custom label if it is defined', () => {
+      aggConfig.params.customLabel = 'Custom label';
+      const label = aggConfig.makeLabel();
+      expect(label).toBe(aggConfig.params.customLabel);
+    });
+
+    it('default label should be "Count"', () => {
+      const label = aggConfig.makeLabel();
+      expect(label).toBe('Count');
+    });
+
+    it('default label should be "Percentage of Count" when percentageMode is set to true', () => {
+      const label = aggConfig.makeLabel(true);
+      expect(label).toBe('Percentage of Count');
+    });
+
+    it('empty label if the type is not defined', () => {
+      aggConfig.type = (undefined as unknown) as AggType;
+      const label = aggConfig.makeLabel();
+      expect(label).toBe('');
+    });
+  });
+
+  describe('#fieldFormatter - custom getFormat handler', () => {
+    it('returns formatter from getFormat handler', () => {
+      setFieldFormats({
+        ...dataPluginMock.createStartContract().fieldFormats,
+        getDefaultInstance: jest.fn().mockImplementation(() => ({
+          getConverterFor: jest.fn().mockImplementation(() => (t: string) => t),
+        })) as any,
+      });
+
+      const ac = new AggConfigs(indexPattern, [], { typesRegistry });
+      const configStates = {
+        enabled: true,
+        type: 'count',
+        schema: 'metric',
+        params: { field: '@timestamp' },
+      };
+      const aggConfig = ac.createAggConfig(configStates);
+
+      const fieldFormatter = aggConfig.fieldFormatter();
+      expect(fieldFormatter).toBeDefined();
+      expect(fieldFormatter('text')).toBe('text');
+    });
+  });
+
+  // TODO: Converting these field formatter tests from browser tests to unit
+  // tests makes them much less helpful due to the extensive use of mocking.
+  // We should revisit these and rewrite them into something more useful.
```

Comment:
Just want to highlight this comment. When these were running as karma tests in the browser, they were really more of integration tests than unit tests. Converting them to unit tests resulted in a ton of mocking, and in the end they aren't quite as useful with mocked field formats. Should probably be improved further down the road.

Reply:
we should just test agg config behaviour when it comes to field formats (so that format is correctly selected) not the actual field formats implementation